src/transformers/trainer.py

```diff
@@ -675,12 +675,14 @@ def train(self, model_path: Optional[str] = None, trial: Union["optuna.Trial", D
 
         # Distributed training (should be after apex fp16 initialization)
         if self.args.local_rank != -1:
+            config = model.config
```

Comment:
We shouldn't assume `model` has a config without proper test, having Traine work with models that are not `PreTrainedModel`s is a feature that has been asked. If there is an access to config that makes the code fail, we should fix that place.

Reply:
Happy to add a check, but the base trainer.py accesses model.config 23 times. This is good IMO, it's the way we control lots of model behavior like dropout, which you might want to change from command line.